### PR TITLE
Fixed navigation bug

### PR DESCRIPTION
**Pull request by Mastermind-sap**
- Also added an exit confirm dialogue box
- Now we can pop until some route so that back button leads to previous page without increasing stack size

### CODE REVIEW

File: lib/pages/login/login_page.dart [hunk 1]
```diff
@@ -23,54 +25,62 @@ class LoginPage extends StatelessWidget {
...
-              "Hey! Welcome",
-              style: Theme.of(context).textTheme.displaySmall,
-            ),
-            FractionallySizedBox(
-              widthFactor: messageFieldWidth,
-              child: Text(
-                "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suscipit sed augue quam amet, sed gravida.",
-                textAlign: TextAlign.center,
-                style: Theme.of(context).textTheme.bodyLarge,
+    return WillPopScope(
```
Comment (Raj217) [inline lib/pages/login/login_page.dart:28]: WillPopScope is depricated try using an alternative of the suggested one of PopScope

File: lib/utils/exit_program.dart [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import 'package:flutter/material.dart';
+
+Future<bool?> showExitWarning(BuildContext context) async {
+  if (Navigator.canPop(context)) {
+    Navigator.pop(context);
+  } else {
+    showDialog<bool>(
```
Comment (Raj217) [inline lib/utils/exit_program.dart]: Not returning the user choice
use return await here to return the value to the previous page and do the necessary

Currently pressing yes is not exiting the app

File: lib/pages/splash_screen/splash_screen.dart [hunk 3]
```diff
@@ -34,8 +34,9 @@ class _SplashScreenState extends State<SplashScreen> {
     init().then((duration) {
       debugPrint("Successfully completed all async tasks");
       debugPrint("Time taken: $duration ms");
-
-      Navigator.pushNamed(context, LoginPage.routeName)
+      Navigator.of(context)
+          .pushNamedAndRemoveUntil(
```
Comment (Raj217) [inline lib/pages/splash_screen/splash_screen.dart:38]: try pushReplacementNamed once

**Review by Raj217 — CHANGES_REQUESTED** [review]

File: lib/utils/exit_program.dart (hunk 2 above)
Comment (Mastermind-sap) [reply lib/utils/exit_program.dart]: Fixed [here](https://github.com/gdsc-nits-org/efficacy_admin/pull/25/commits/e305f928eca46451eca64c901c3a38941f44b254)

**Review by Raj217 — APPROVED** [review]

**Pull request merged**